### DISCUSSION TITLE
Binding to a date attribute in an Eloquent Model when in a different timezone than UTC

### DISCUSSION THREAD

**Discussion by oxytoxin** [livewire/livewire#3247] [7. Bugs (archived)] [answered]
### Discussed in https://github.com/livewire/livewire/discussions/3212

<div type='discussions-op-text'>

<sup>Originally posted by **oxytoxin** July  1, 2021</sup>
I have an eloquent model called Task that has a deadline column which is included in the model's $dates attribute array. My timezone in the config files is set to Asia/Manila.
I have a livewire component that binds to this Task model using wire:model. The problem I encountered was that everytime a livewire action gets called, e.g. via a button click that triggers an action, the model's deadline attribute gets readjusted.

To be more precise, it seems that the deadline's value is taken as having a UTC timezone and gets readjusted to my set timezone in the config files every time livewire makes a roundtrip. Does this perhaps have something to do with the model's Serialization & Deserialization?

![image](https://user-images.githubusercontent.com/61694153/124005397-9ab54b00-da0b-11eb-9163-c798e590ac1e.png)
![image](https://user-images.githubusercontent.com/61694153/124005471-b4569280-da0b-11eb-91e1-289530574630.png)

This is what it looks like before pressing a button
![image](https://user-images.githubusercontent.com/61694153/124005712-f97ac480-da0b-11eb-9d15-4564f0342f71.png)
And after pressing validate
![image](https://user-images.githubusercontent.com/61694153/124005753-07304a00-da0c-11eb-813f-a08ced6bd753.png)

Thank you for your help! Cheers!</div>

**Comment by marksalmon** [comment, marked as answer]
I had the exact same problem.

Casting to an immutable date seems to fix it

```php
protected $casts = [
    'deadline' => 'date_immutable',
    'opening_on' => 'date_immutable'
];
```

**Reply by oxytoxin** [reply]
You're right, it seems to be the case. Thank you!
